tests/validateManifest.test.ts

```diff
@@ -0,0 +1,95 @@
+import { RuleTester } from "@typescript-eslint/rule-tester";
+import manifestRule from "../lib/rules/validateManifest.js";
+
+const ruleTester = new RuleTester();
+
+ruleTester.run("validate-manifest", manifestRule, {
+	valid: [
+		{
+			filename: "manifest.json",
+			code: `{
+                    "id": "my-plugin",
```

Comment:
This should also be invalid, including the word `plugin` in the id and name is discouraged

Reply:
I have replaced the check for `Obsidian` with an extensible banned word list. (currently contains `obsidian` and `plugin`)